.gitignore

```diff
@@ -3,6 +3,7 @@
 .settings
 build
 build.ninja
+build.ninja.new
```

Comment:
yeah, but it's not a new thing actually. it's created by https://github.com/scylladb/scylladb/blob/f7d5d4877c360fff250ef2ef0e5301a23eeb2642/configure.py#L2394 when reconfiguring the building system and updating `build.ninja`, we 

1. create the new `build.ninja` with the file name of `build.ninja.new`, and
2. replace the old one with the new one.

if your repo has it, there are good chance that the above steps were interrupted or `configure.py` failed in the middle.

Reply:
I also have this `build.ninja.new` file on disk. It has this interesting part:

```
rule configure
  command = python3 configure.py --out=build.ninja.new.new.new $configure_args && mv build.ninja.new.new.new build.ninja.new.new
  generator = 1
  description = CONFIGURE $configure_args
```

I suppose that when ninja decides that reconfigure is needed, it creates the `build.ninja.new` file and then renames it, but when it generates the file it uses `./configure.py --out=build.ninja.new`, so the reconfigure command in the new file will have the `.new` suffix appended twice.

My theory is:
- A developer runs `./configure.py`. It generates a `build.ninja` file. It contains the following reconfigure command: `python3 configure.py --out=build.ninja.new && mv build.ninja.new build.ninja`
- During build, ninja decides that it needs to reconfigure. It calls the reconfigure command, generates `build.ninja.new` and renames to `build.ninja`. However, the reconfigure command in the new file is `python3 configure.py --out=build.ninja.new.new && mv build.ninja.new.new build.ninja.new`
- During build, ninja once again decides to reconfigure. It calls the reconfigure command, generates `build.ninja.new.new` and renames to `build.ninja.new`. The `build.ninja` file is not overwritten, and the new `build.ninja.new` file should look like the one I have shown above.

Reply:
@nuivall could you check whether your `build.ninja.new` file also has these long repeated `.new` suffixes in the `configure` rule? If so then it's probably the same problem.

I think we should add an argument to `configure.py`, specific to the reconfigure case, which tells the script the name of the file that the generated file is supposed to be overwriting. The code which outputs the `rule configure` stuff should be adjusted to use that.

Reply:
Yeah, it's pretty easy to reproduce:
- Start with a fresh repo
- Run `./configure.py` -> generates `build.ninja` with `mv build.ninja.new build.ninja` inside it
- Build the project
- `touch configure.py`
- Build the project -> for some reason `./configure.py` runs twice, so this already creates both the broken `build.ninja` and the additional `build.ninja.new` in the way I described.

I'll go ahead and prepare/send a fix.

Reply:
Sent a fix: https://github.com/scylladb/scylladb/pull/19428

As for the PR this conversation was originally about - perhaps it is still useful? The `build.ninja.new` file can still be created if you interrupt the ninja makefile regeneration. @tchaikov @nuivall WDUT?

Reply:
I think if it's still something that potentially can be created, and we don't care about it (i.e. it doesn't make build incorrect or something) then we still can merge it to avoid annoyance of git clean

Reply:
> Sent a fix: #19428
> 
> As for the PR this conversation was originally about - perhaps it is still useful? The `build.ninja.new` file can still be created if you interrupt the ninja makefile regeneration. @tchaikov @nuivall WDUT?

Actually, the whole reason for this "build.ninja.new" trick was that on interrupt, ninja will delete this file so we wanted it to delete build.ninja.new and not our original build.ninja ;-) But I guess it is true that if you're very unlucky, suffer a power outage or SIGKILL or something, you can indeed find yourself with that build.ninja.new remaining. So I'm not against this patch.